dashboard/app/controllers/levels_controller.rb

```diff
@@ -7,7 +7,7 @@ class LevelsController < ApplicationController
   before_filter :authenticate_user!, :except => [:embed_blocks], :unless => Proc.new {params[:embed] && action_name == 'show'}
```

Comment:
These exceptions should also no longer be needed if these actions are moved from this controller.